include/prometheus/gateway.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <memory>
+#include <sstream>
+
+#include <cpr/cpr.h>
+
+#include "histogram.h"
+#include "registry.h"
+
+namespace prometheus {
+
+class Gateway {
+ public:
+  typedef std::map<std::string, std::string> labels_t;
+
+  explicit Gateway(const std::string& uri, const std::string jobname,
+                   const labels_t* labels = nullptr,
+                   const std::string username = "",
+                   const std::string password = "");
+  ~Gateway() {}
+
+  void RegisterCollectable(const std::weak_ptr<Collectable>& collectable,
+                           const labels_t* labels = nullptr);
+
+  static const labels_t instance_label(void);
+
+  // Push metrics to the given pushgateway.
+  int Push(void) { return push(false); }
+
+  std::future<int> AsyncPush(void) { return async_push(false); }
+
+  // PushAdd metrics to the given pushgateway.
+  int PushAdd(void) { return push(true); }
+
+  std::future<int> AsyncPushAdd(void) { return async_push(true); }
+
+  // Delete metrics from the given pushgateway.
+  int Delete(void);
```

Comment:
Please remove the void, it doesn't make any difference in C++ (as opposed to C).

Reply:
fixed in #124